notebooks/ch16/classification_cost.md

```diff
@@ -195,4 +205,55 @@ The surprisingly simple gradient expression allows us to fit a logistic model to
+
+## KL Divergence and Cross-Entropy Cost
+
+### Defining KL Divergence
+
+As noted at the beginning of the section, the goal in our binary classification example is to closely approximate the real data distribution using a logistic model with parameters $\theta$:
+
+$$ \hat{P_\theta}(y = 1 | x) \approx P(y = 1 | x) $$
+
+**Kullback–Leibler (KL) divergence** provides a measure of the difference between two probability distributions. It reveals how imprecise the probability distribution output by a logistic regression model is relative to the data. KL-divergence, also called relative entropy, has its roots in information theory (beyond the scope of DS 100).
+
+The KL divergence between two distributions $P$ and $\hat{P_\theta}$ for a single data point $(x, y)$ is given by:
+
+$$D(P || \hat{P_\theta}) = \sum_{k=0}^{K-1} P(y = k | x) \ln \left(\frac{P(y = k | x)}{\hat{P_\theta}(y = k | x)}\right)$$
+
+$K$ represents the number of possible outcomes, so in the case of binary classification $K = 2$. 
+
+The best $\theta$ minimizes the average KL divergence of the entire dataset of $n$ points:
+
+$$\displaystyle\arg \min_{\substack{\theta}} \frac{1}{n} \sum_{i=1}^{n} \sum_{k=0}^{K-1} P(y_i = k | x_i) \ln \left(\frac{P(y_i = k | x_i)}{\hat{P_\theta}(y_i = k | x_i)}\right)$$
+
+Average KL divergence can be interpreted as the average log difference between the two distributions $P$ and $\hat{P_\theta}$ weighted by $P$. The structure of the equation has two important implications:
+
+* KL divergence does not penalize mismatch for rare events with respect to $P$. If the model predicts a high probability for an event that is actually rare, then both $P(k)$ and $\ln \left(\frac{P(k)}{\hat{P}(k)}\right)$ are low so the divergence is also low. However, if the model predicts a low probability for an event that is actually common, then the divergence is high. 
+* KL divergence is not symmetric: $$D(P || \hat{P_\theta}) \neq D(\hat{P_\theta} || P)$$ In other words, the divergence of $\hat{P_\theta}$ from $P$ is not the same as the divergence of $P$ from $\hat{P_\theta}$.
+
+We can deduce that a logistic regression model that accurately predicts common events has a lower divergence from the real distribution than does a model that accurately predicts rare events but varies widely on common events. 
+
+### Deriving Cross-Entropy Cost from KL Divergence
+
+The logarithm of the ratio of probabilities in the average KL divergence equation hints back to our study of the cross-entropy cost equation. In fact, minimizing average KL divergence is equivalent to minimizing average cross-entropy cost. The rest of this section walks through an algebraic derivation of cross-entropy cost from average KL divergence.
```

Comment:
Since I think we should mention that minimizing KL divergence is equivalent to minimizing cross-entropy earlier in the section, the second sentence should be reworded or omitted.

nit: Let's reword the last sentence to "We will demonstrate this equivalence by deriving the cross-entropy cost from the KL divergence mathematically."